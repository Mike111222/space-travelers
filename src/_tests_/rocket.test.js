import { render } from '@testing-library/react';
import Rocket from '../components/rocket';

describe('tests for rocket components', () => {
  test('check snapshot for rocket', () => {
    const data = [
      {
        id: 1,
        name: 'rocket 1',
        image: 'image1',
        description: 'description of rocket1',
        reserved: false,
      },
      {
        id: 2,
        name: 'rocket 2',
        image: 'image2',
        description: 'description of rocket2',
        reserved: false,
      },
    ];
    const Reservation = jest.fn();
    const Cancelation = jest.fn();

    const tree = render(
      <Rocket
        data={data}
        Reservation={Reservation}
        Cancelation={Cancelation}
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
