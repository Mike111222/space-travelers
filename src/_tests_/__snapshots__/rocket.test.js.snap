// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rocket components should match snapshot for rockets  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rocketDisplay"
      >
        <img
          alt="rocket 1's display"
          class="rocketImage"
          src="image1"
        />
        <section
          class="rInfo"
        >
          <span
            class="title"
          >
            rocket 1
          </span>
          <p
            class="rocketpara"
          >
            description of rocket1
          </p>
          <button
            class="book"
            type="submit"
          >
            Reserve Rocket
          </button>
        </section>
      </div>
      <div
        class="rocketDisplay"
      >
        <img
          alt="rocket 2's display"
          class="rocketImage"
          src="image1"
        />
        <section
          class="rInfo"
        >
          <span
            class="title"
          >
            rocket 2
          </span>
          <p
            class="rocketpara"
          >
            description of rocket1
          </p>
          <button
            class="book"
            type="submit"
          >
            Reserve Rocket
          </button>
        </section>
      </div>
  </body>,
  "container": <div>
    <div
      class="rocketDisplay"
    >
      <img
        alt="rocket 1's display"
        class="rocketImage"
        src="image1"
      />
      <section
        class="rInfo"
      >
        <span
          class="title"
        >
          rocket 1
        </span>
        <p
          class="rocketpara"
        >
          description of rocket1
        </p>
        <button
          class="book"
          type="submit"
        >
          Reserve Rocket
        </button>
      </section>
    </div>
    <div
      class="rocketDisplay"
    >
      <img
        alt="rocket 2's display"
        class="rocketImage"
        src="image2"
      />
      <section
        class="rInfo"
      >
        <span
          class="title"
        >
          rocket 2
        </span>
        <p
          class="rocketpara"
        >
          description of rocket2
        </p>
        <button
          class="book"
          type="submit"
        >
          Reserve Rocket
        </button>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`tests for rocket components check snapshot for rocket 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rocketDisplay"
      >
        <img
          alt="rocket undefineds display"
          class="rocketImage"
        />
        <section
          class="rInfo"
        >
          <span
            class="title"
          />
          <p />
          <button
            class="book"
            type="submit"
          >
            Reserve Rocket
          </button>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rocketDisplay"
    >
      <img
        alt="rocket undefineds display"
        class="rocketImage"
      />
      <section
        class="rInfo"
      >
        <span
          class="title"
        />
        <p />
        <button
          class="book"
          type="submit"
        >
          Reserve Rocket
        </button>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
